aws_lambda_powertools/event_handler/api_gateway.py

```diff
@@ -670,6 +674,9 @@ def _openapi_operation_metadata(self, operation_ids: set[str]) -> dict[str, Any]
         operation_ids.add(self.operation_id)
         operation["operationId"] = self.operation_id
 
+        # Mark as deprecated if necessary
+        operation["deprecated"] = self.deprecated or None
```

Comment:
I simplified it by adding `self.depecrated or None` because we don't serialize `None` values, so it won't appear in the schema definition if it is `False`.